From 6c341544e189b91ab9366d17aa73eeb09bfae52d Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Wed, 18 Jan 2012 10:37:55 +0100
Subject: [PATCH 01/52] Simplify qemu_realloc()

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1326883126-22053-2-git-send-email-armbru@redhat.com>
Patchwork-id: 36577
O-Subject: [RHEL-6.3 PATCH qemu-kvm 01/52] Simplify qemu_realloc()
Bugzilla: 758194
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>

No functional change.  Bonus: looks just like qemu_malloc() now.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>
(cherry picked from commit f8b0953bd5e34f87458108051a485246d40adadc)

Conflicts:

	qemu-malloc.c

Conflicts because commit d56025ec went in out of order.
---
 qemu-malloc.c |   14 +++++---------
 1 files changed, 5 insertions(+), 9 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 qemu-malloc.c |   14 +++++---------
 1 files changed, 5 insertions(+), 9 deletions(-)

diff --git a/qemu-malloc.c b/qemu-malloc.c
index bd82e6f..72de60a 100644
--- a/qemu-malloc.c
+++ b/qemu-malloc.c
@@ -67,16 +67,12 @@ void *qemu_malloc(size_t size)
 void *qemu_realloc(void *ptr, size_t size)
 {
     void *newptr;
-    if (size) {
-        newptr = oom_check(realloc(ptr, size));
-        trace_qemu_realloc(ptr, size, newptr);
-        return newptr;
-    } else if (allow_zero_malloc()) {
-        newptr = oom_check(realloc(ptr, size ? size : 1));
-        trace_qemu_realloc(ptr, size, newptr);
-        return newptr;
+    if (!size && !allow_zero_malloc()) {
+        abort();
     }
-    abort();
+    newptr = oom_check(realloc(ptr, size ? size : 1));
+    trace_qemu_realloc(ptr, size, newptr);
+    return newptr;
 }
 
 void *qemu_mallocz(size_t size)
-- 
1.7.7.5

